stablehlo/tests/BUILD.bazel

```diff
@@ -134,8 +134,8 @@ expand_template(
     testonly = True,
     out = "lit.site.cfg.py",
     substitutions = {
-        "\"@STABLEHLO_SOURCE_DIR@\"": "os.path.join(os.environ['TEST_SRCDIR'], 'stablehlo')",
-        "\"@STABLEHLO_TOOLS_DIR@\"": "os.path.join(os.environ['TEST_SRCDIR'], 'stablehlo')",
+        "\"@STABLEHLO_SOURCE_DIR@\"": "os.path.join(os.environ['TEST_SRCDIR'], '_main')",
+        "\"@STABLEHLO_TOOLS_DIR@\"": "os.path.join(os.environ['TEST_SRCDIR'], '_main')",
```

Comment:
Just a fly-by comment: I think this might break some of our downstream integrations pulling in stablehlo (and running its tests as part of downstream CI). A thought: Would adding `common --noenable_bzlmod` to the `.bazelrc` (like [this](https://github.com/cruise-automation/mlir-tcp/blob/main/.bazelrc#L48-L50)) resolve this, at least until stablehlo actually migrates to using bzlmod (depends on llvm migrating over as you clarified)?

Reply:
I am playing around with:

```
❯ cat s\tablehlo/tests/utils.bzl

def _is_bzlmod_enabled():
    """Determine whether bzlmod mode is enabled."""
    # If bzlmod is enabled, then `str(Label(...))` returns a canonical label,
    # these start with `@@`.
    return str(Label("//:invalid")).startswith("@@")

def workspace_name():
    """Return the name of the workspace."""
    return "_main" if _is_bzlmod_enabled() else "stablehlo"
```

```
diff --git a/stablehlo/tests/BUILD.bazel b/stablehlo/tests/BUILD.bazel
index 8ce10114..d8b3046a 100644
--- a/stablehlo/tests/BUILD.bazel
+++ b/stablehlo/tests/BUILD.bazel
@@ -15,6 +15,7 @@ load("@bazel_skylib//rules:expand_template.bzl", "expand_template")
 load("@llvm-project//llvm:lit_test.bzl", "lit_test", "package_path")
 load("@llvm-project//mlir:tblgen.bzl", "gentbl_cc_library", "td_library")
 load("@rules_cc//cc:defs.bzl", "cc_library")
+load("//stablehlo/tests:utils.bzl", "workspace_name")
 
 package(
     default_visibility = ["//visibility:public"],
@@ -134,7 +135,7 @@ expand_template(
     testonly = True,
     out = "lit.site.cfg.py",
     substitutions = {
-        "\"@STABLEHLO_SOURCE_DIR@\"": "os.path.join(os.environ['TEST_SRCDIR'], '_main')",
+        "\"@STABLEHLO_SOURCE_DIR@\"": "os.path.join(os.environ['TEST_SRCDIR'], '{}')".format(workspace_name()),
         "\"@STABLEHLO_TOOLS_DIR@\"": "os.path.join(os.environ['TEST_SRCDIR'], '_main')",
         "@LIT_SITE_CFG_IN_HEADER@": "# Autogenerated, do not edit.",
         "@LLVM_TOOLS_DIR@": package_path("@llvm-project//llvm:BUILD"),
```

Reply:
> having to keep our Bazel files sync'd with every downstream seems untenable though; it means if anyone bumps we have to do it in lockstep which doesn't really make sense.

Using `workspace_name` in the paths made it so it worked in both scenarios: 1) stablehlo's standalone builds, and 2) downstream in a monorepo setting where stablehlo is added (as an `http_archive`). I'm just pointing out that the change in this PR would break 2 for all the downstream repos, and your patch on conditionally switching paths depending on `_is_bzlmod_enabled` seems to mitigate that. For context, we used to maintain a patch file for stablehlo integrations, which we could entirely retire after https://github.com/openxla/stablehlo/pull/1810.

Reply:
Let me send up a patch PR for this so that we have backwards compatability.

Reply:
@sjain-stanford I also have this tested and working. I'm not sure which is more idiomatic.

```

# Equivalent of configure_lit_site_cfg from CMakeLists.txt.
expand_template(
    name = "lit_site_cfg_py_gen",
    testonly = True,
    out = "lit.site.cfg.py",
    substitutions = {
        "\"@STABLEHLO_SOURCE_DIR@\"": "RUNFILES_DIR",
        "\"@STABLEHLO_TOOLS_DIR@\"": "RUNFILES_DIR",
        "@LIT_SITE_CFG_IN_HEADER@": '''# Autogenerated, do not edit.
from python.runfiles import Runfiles
from pathlib import Path

r = Runfiles.Create()
LITE_CFG_PY = Path(r.Rlocation("stablehlo/stablehlo/tests/lite.cfg.py"))
RUNFILES_DIR = LITE_CFG_PY.parent.parent.parent.absolute().as_posix()''',
        "@LLVM_TOOLS_DIR@": package_path("@llvm-project//llvm:BUILD"),
    },
    template = "lit.site.cfg.py.in",
)
```

This uses the @python_rules and then walks up the directory from a known path.

Reply:
@fzakaria thanks. If you share a branch, I'd be happy to test it from our monorepo.